forwardproxy.go

```diff
@@ -38,13 +38,27 @@ type ForwardProxy struct {
 	authCredentials    [][]byte // slice with base64-encoded credentials
 	hideIP             bool
 	hideVia            bool
+	useProxy           bool
 	whitelistedPorts   []int
 	probeResistDomain  string
 	pacFilePath        string
 	probeResistEnabled bool
 	dialTimeout        time.Duration // for initial tcp connection
 	hostname           string        // do not intercept requests to the hostname (except for hidden link)
 	port               string        // port on which chain with forwardproxy is listening on
+	outgoing           struct {
+		IPs    []string
+		policy Policy
+	}
+}
+
+var (
+	supportedPolicies = make(map[string]func(string) Policy)
+)
+
+// RegisterPolicy adds a custom policy to the forward proxy.
+func RegisterPolicy(name string, policy func(string) Policy) {
```

Comment:
How do we expect this to be used?
If it ultimately requires to modify source code and build from it, doesn't seem to add that much flexibility, since one can do whatever they want at this point. Perhaps, I am missing something.